src/main/scala/strawman/collection/View.scala

```diff
@@ -154,14 +154,14 @@ object View extends IterableFactory[View] {
     def iterator() = underlying.iterator().flatMap(f)
   }
 
-  /** A view that concatenates elements of the underlying collection with the elements
-   *  of another collection or iterator.
+  /** A view that concatenates elements of the prefix collection or iterator with the elements
+   *  of the suffix collection or iterator.
    */
-  case class Concat[A](underlying: Iterable[A], other: IterableOnce[A]) extends View[A] {
-    def iterator() = underlying.iterator() ++ other
-    override def knownSize = other match {
-      case other: Iterable[_] if underlying.knownSize >= 0 && other.knownSize >= 0 =>
-        underlying.knownSize + other.knownSize
+  case class Concat[A](prefix: IterableOnce[A], suffix: IterableOnce[A]) extends View[A] {
+    def iterator() = prefix.iterator() ++ suffix.iterator()
```

Comment:
The fact that `View` extends `Iterable` (and not just `IterableOnce`) means that it should be possible to repeatedly iterate on a same view. However, since we accept `IterableOnce`s as parameters, this promise doesn’t hold anymore. I’m wondering if that could be the source of subtle bugs…

Note that it was already the case before your PR.